jxl/src/entropy_coding/ans.rs

```diff
@@ -2,6 +2,486 @@
+        if shift > 13 {
+            return Err(Error::InvalidAnsHistogram);
+        }
+
+        let alphabet_size = Self::read_u8(br)? as usize + 3;
+        if alphabet_size > table_size {
+            return Err(Error::InvalidAnsHistogram);
+        }
+
+        // TODO(tirr-c): This could be an array of length `SUM_PROB / 4` (4 is from the minimum
+        // value of `repeat_count`). Change if using array is faster.
+        let mut repeat_ranges = Vec::new();
+        let mut omit_data = None;
+        let mut idx = 0;
+        while idx < alphabet_size {
+            dist[idx] = Self::read_prefix(br)?;
+            if dist[idx] == RLE_MARKER_SYM {
+                let repeat_count = Self::read_u8(br)? as usize + 4;
+                if idx + repeat_count > alphabet_size {
+                    return Err(Error::InvalidAnsHistogram);
+                }
+                repeat_ranges.push(idx..(idx + repeat_count));
+                idx += repeat_count;
+                continue;
+            }
+            match &mut omit_data {
+                Some((log, pos)) => {
+                    if dist[idx] > *log {
+                        *log = dist[idx];
+                        *pos = idx;
+                    }
+                }
+                data => {
+                    *data = Some((dist[idx], idx));
+                }
+            }
+            idx += 1;
+        }
+        let Some((_, omit_pos)) = omit_data else {
+            return Err(Error::InvalidAnsHistogram);
+        };
+        if dist.get(omit_pos + 1) == Some(&RLE_MARKER_SYM) {
+            return Err(Error::InvalidAnsHistogram);
+        }
+
+        let mut repeat_range_idx = 0usize;
+        let mut acc = 0;
+        let mut prev_dist = 0u16;
+        for (idx, code) in dist.iter_mut().enumerate() {
+            if repeat_range_idx < repeat_ranges.len()
+                && repeat_ranges[repeat_range_idx].start <= idx
+            {
+                if repeat_ranges[repeat_range_idx].end == idx {
+                    repeat_range_idx += 1;
+                } else {
+                    *code = prev_dist;
+                    acc += *code;
+                    if acc > SUM_PROBS {
+                        return Err(Error::InvalidAnsHistogram);
+                    }
+                    continue;
+                }
+            }
+
+            if *code == 0 {
+                prev_dist = 0;
+                continue;
+            }
+            if idx == omit_pos {
+                prev_dist = 0;
+                continue;
+            }
+            if *code > 1 {
+                let zeros = (*code - 1) as i16;
+                let bitcount = (shift - ((LOG_SUM_PROBS as i16 - zeros) >> 1)).clamp(0, zeros);
+                *code = (1 << zeros) + ((br.read(bitcount as usize)? as u16) << (zeros - bitcount));
+            }
+            prev_dist = *code;
+            acc += *code;
+            if acc > SUM_PROBS {
```

Comment:
Shouldn't this be `>=`? I don't think it is valid to have the omit_pos symbol have a probability of 0...

Reply:
The spec says `D[omit_pos] >= 0`, but yeah, omit symbol having a probability of 0 doesn't make much sense... Is this a spec bug?

Reply:
Interesting, libjxl definitely checks for `D[omit_pos] > 0`: https://github.com/libjxl/libjxl/blob/main/lib/jxl/dec_ans.cc#L178C1-L183C1

@jonsneyers thoughts?

Reply:
So this is either a spec bug or libjxl being too strict. If we follow the spec, what happens? Having `D[omit_pos] == 0` is never useful, right? It's just a non-optimal way of writing a distribution.
I think it's best to see it as a spec bug and update the spec to make the libjxl behavior correct.

Reply:
I think I agree with Jon here - let's call this a spec bug, and check for strictly-positive (i.e. `acc >= SUM_PROBS`)